includes/class-wc-payments-incentives-service.php

```diff
@@ -0,0 +1,208 @@
+	}
+
+	/**
+	 * Add connect incentive to JS settings.
+	 *
+	 * @param array $settings Current JS settings.
+	 * @return array Updated JS settings.
+	 */
+	public function add_js_settings( array $settings ): array {
+		$settings['connectIncentive'] = $this->get_cached_connect_incentive();
+
+		return $settings;
+	}
+
+	/**
+	 * Add badge to payments menu if there is an eligible incentive.
+	 *
+	 * @return void
+	 */
+	public function add_payments_menu_badge(): void {
+		global $menu;
+
+		if ( ! $this->get_cached_connect_incentive() ) {
+			return;
+		}
+
+		foreach ( $menu as $index => $menu_item ) {
+			if ( 'wc-admin&path=/payments/connect' === $menu_item[2] ) {
+				$menu[ $index ][0] .= WC_Payments_Admin::MENU_NOTIFICATION_BADGE; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+				break;
+			}
+		}
+	}
+
+	/**
+	 * Adds allowed promo notes from eligible incentive.
+	 *
+	 * @param array $promo_notes Current allowed promo notes.
+	 * @return array Updated allowed promo notes.
+	 */
+	public function allowed_promo_notes( $promo_notes = [] ): array {
+		if ( $this->get_cached_connect_incentive() ) {
+			/**
+			 * Suppress psalm error because we already check that `id` exists in `is_valid_cached_incentive`.
+			 *
+			 * @psalm-suppress PossiblyNullArrayAccess */
+			$promo_notes[] = $this->get_cached_connect_incentive()['id'];
+		}
+
+		return $promo_notes;
+	}
+
+	/**
+	 * Gets and caches eligible connect incentive from the server.
+	 *
+	 * @param bool $force_refresh Forces data to be fetched from the server, rather than using the cache.
+	 *
+	 * @return array|null List of incentives or null.
+	 */
+	public function get_cached_connect_incentive( bool $force_refresh = false ): ?array {
+		// Return early if there is an account connected.
+		if ( WC_Payments::get_account_service()->is_stripe_connected() ) {
+			return null;
+		}
+
+		// Return early if the country is not supported.
+		if ( ! array_key_exists( WC()->countries->get_base_country(), WC_Payments_Utils::supported_countries() ) ) {
+			return null;
+		}
+
+		$incentive = $this->database_cache->get_or_add(
+			Database_Cache::CONNECT_INCENTIVE_KEY,
+			[ $this, 'fetch_connect_incentive' ],
+			'is_array',
+			$force_refresh,
+		);
+
+		if ( ! $this->is_valid_cached_incentive( $incentive ) ) {
+			return null;
+		}
+
+		return $incentive;
+	}
+
+	/**
+	 * Fetches eligible connect incentive from the server.
+	 *
+	 * @return array|null Array of eligible incentive data or null.
+	 */
+	public function fetch_connect_incentive(): ?array {
+		// Request incentive from WCPAY API.
+		$url = add_query_arg(
+			[
+				// Store ISO-2 country code, e.g. `US`.
+				'country'      => WC()->countries->get_base_country(),
+				// Store locale, e.g. `en_US`.
+				'locale'       => get_locale(),
+				// WCPay active for duration in seconds.
+				'active_for'   => time() - get_option( 'wcpay_activation_timestamp', time() ),
```

Comment:
We should keep the meaning of `active_for` to be store lifetime (the same way as in WC core), not since WCPay activation. We would complicate the server logic with different meanings behind these signals.

If we want to know how long WCPay has been active we can add a new query arg (like `wcpay_active_for`). The `/incentives` endpoint will decide if it uses it or not.

Reply:
For this case we don't actually need that information. But as the name is just `active_for` and we will act differently based on the `caller`, I don't see a need for another param.

A similar case to `has_wcpay`, we can treat them differently based on the `caller` or add new params. They sound generic enough to me to use the same.

Reply:
To be able to merge it already, I reverted `active_for` to WC time. As we don't need it for the eligibility in this case, we can decide later about how to treat it 🙂 And pretty much the same applies to `has_wcpay`.

Reply:
Excellent! We may never know when we want to take into account the store age, especially with the shifting requirements.